packages/frontend/app/(dashboard)/organization/settings/page.tsx

```diff
@@ -30,7 +30,26 @@ import { useOrganizationSettings } from '@/app/hooks/useOrganizationSettings'
 import { checkCourseCreatePermissions } from '@/app/utils/generalUtils'
 
 export default function SettingsPage(): ReactElement {
+  // Handler to update SSO patterns
+  const updateSSO = async () => {
+    try {
+      const formValues = await formSSO.validateFields()
+      const ssoEmailPatterns = formValues.ssoEmailPatterns || []
+      await API.organizations.patch(organization?.id ?? -1, {
+        ssoEmailPatterns,
+      })
+      message.success('SSO patterns updated')
+      setTimeout(() => {
+        window.location.reload()
+      }, 1750)
```

Comment:
```suggestion
  setUserInfo({
  ...userInfo,
  userInfo.org: {
  ...org,
  ssoEmailPatterns: ssoEmailPatterns
  }
  })
```
I know there's another instance of this "set a timeout to refresh the page" code on this page, but it should properly update the local state variable. For context, the page refresh was done by another developer while I was first starting. They didn't really know how to update the local userInfo context so they just triggered a page refresh lol. They did the same on the profile page, though I hadn't gotten around to giving this page love yet.

Also the formatting and syntax might be a bit scuffed sorry.

You will also need to adjust the code below so that you can access the setUserInfo method.
```
const { userInfo, setUserInfo } = useUserInfo()
```

And maybe, if you have time, you could probably update the other window.location.reload() stuff to instead update the userInfo context with the correct org info.

Reply:
oh god i just looked at this old code below more, *each form value has its own state variable* that's *so* gross (the `form` state already has all of these and is handled by antd). It also has a bunch of checks inside the submit function instead of using a validator function for the component (so the user would instantly get feedback about something being wrong instead of waiting until they click submit). I guess the person never read the antd docs for creating forms